### ISSUE TITLE
Charting dependencies

### ISSUE THREAD

**Issue by rishigurnani** [rusty1s/pytorch_geometric#2266]
From now on, we recommend using our discussion forum (https://github.com/rusty1s/pytorch_geometric/discussions) for general questions.

## ❓ Questions & Help

It looks like information in my custom Message Passing layer is being passed along the batch dimension, which should not happen. I create a loss function which is just equal to the output of the first connected graph in the batch. But when I backprop all the way back to the input node features, nodes that are not in the first graph have non-zero gradients.

**Comment by rishigurnani** [comment]
If I share the custom layer (PseudoDC) is that helpful?

```
from torch_scatter import scatter_sum, scatter_mean
from torch_sparse import SparseTensor
from torch import Tensor
from torch_geometric.nn.conv import GatedGraphConv as GGC
import torch.nn as nn
import torch
import torch.nn.functional as F
from torch_geometric.nn import MessagePassing
import random
import numpy as np

random.seed(2)
torch.manual_seed(2)
np.random.seed(2)

class PseudoDC(MessagePassing): #Pseudo DeepChem layer
  def __init__(self,debug=False):
    super(PseudoDC, self).__init__(aggr='add',node_dim=0)
    self.debug = debug
    self.V = my_hidden(70, 70, None)
    self.U = my_hidden(146, 70, F.elu)
    self.E = my_hidden(6, 6, None)


  def forward(self,x, edge_index,edge_weight, batch):
    if self.debug:
      print('#####')
    return self.propagate(edge_index=edge_index,x=x,edge_weight=edge_weight,batch=batch)
  
  def update(self,aggr_out,x):
    '''
    aggr_out = output of self.aggregate( self.message(.) ) 
    '''
    if self.debug:
        print('x:',x[:,0:10])
        print('aggr_out:',aggr_out[:,0:10])
        pass #do not comment out

    up = F.elu( torch.cat( (aggr_out, x),1 ) ) 
    out = self.U(up)
    return out

  def message(self,x_i,x_j,edge_weight,edge_index):
    '''
    x_j are the neighboring atoms that correspond to x 
    '''
    if self.debug:
      print(edge_index)
      print('x_i:', x_i[:,0:10])
      print('x_j:', x_j[:,0:10])
      pass #do not comment out
    
    if self.debug:
      return torch.cat( (x_j, edge_weight), 1)
    
    m_j = self.V(x_j)
    m_ij = self.E(edge_weight)
    return torch.cat( (m_j, m_ij), 1)

class my_hidden(nn.Module):
    '''
    Hidden layer with xavier initialization and batch norm
    '''
    def __init__(self, size_in, size_out, activation):
        super().__init__()    
        self.size_in, self.size_out = size_in, size_out
        self.linear = nn.Linear(self.size_in,self.size_out)
        nn.init.xavier_uniform(self.linear.weight)
        self.activation = activation
        self.bn = nn.BatchNorm1d(self.size_out)
    
    def forward(self,x):
        if self.activation is not None:
            return self.bn( self.activation( self.linear(x) ) )
        else:
            return self.bn( self.linear(x) )
```

**Comment by rusty1s** [comment]
Are you sure this is not induced by `BatchNorm`?

**Comment by rishigurnani** [comment]
Good point! I removed `BatchNorm` now but the issue still persists.

![image](https://user-images.githubusercontent.com/13811245/111538005-85f5c580-8742-11eb-9605-3f102f5f1e4a.png)


Here is how I'm getting the gradient of `x`.

![image](https://user-images.githubusercontent.com/13811245/111537836-4d55ec00-8742-11eb-95dc-acf4a3b782e6.png)

Note: `min_model` uses the `PseudoDC` layer.

**Comment by rishigurnani** [comment]
Never mind, I found there was another layer which had batch norm. When I removed it the problem is solved.

Thank you!
